### PR TITLE
Add SDK to Get Current Period from the API

### PR DESCRIPTION
**Pull request by BSaunders95**
Adds use of the SDK to retrieve the current period for the balance sheet.

Note that once approved, this branch will only be merged following the merge of PR: https://github.com/companieshouse/company-accounts.api.ch.gov.uk/pull/41 which exposes the GET endpoint in the API.

Required for SFA-596

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java [hunk 1]
```diff
@@ -12,14 +16,22 @@
     @Autowired
     private BalanceSheetTransformer transformer;
 
+    @Autowired
+    private ApiClientService apiClientService;
+
     @Override
-    public BalanceSheet getBalanceSheet(String transactionId, String companyAccountsId) {
+    public BalanceSheet getBalanceSheet(String transactionId, String companyAccountsId)
+            throws ApiErrorResponseException {
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java]: This is inconsistent with how we are handling errors in the [API services](https://github.com/companieshouse/company-accounts.api.ch.gov.uk/blob/develop/src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java). We have separated the knowledge bleed between layer so the controller doesn't know the internal behaviour of the service (making an API call).

File: src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BalanceSheetController.java [hunk 2]
```diff
@@ -27,7 +28,12 @@ public String getBalanceSheet(@PathVariable String transactionId,
                                   @PathVariable String companyAccountsId,
                                   Model model) {
 
-        model.addAttribute("balanceSheet", balanceSheetService.getBalanceSheet(transactionId, companyAccountsId));
+        try {
+            model.addAttribute("balanceSheet", balanceSheetService.getBalanceSheet(transactionId, companyAccountsId));
+        } catch (ApiErrorResponseException e) {
+            // TODO: Handle with exception handler once implemented
+            e.printStackTrace();
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BalanceSheetController.java]: This should be using structured logging (if it gets missed in a future PR it won't be logging in the correct format).

Comment (BSaunders95) [reply src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/BalanceSheetController.java]: Used the Structured Logging library for `LOGGER.error()` in place of printing the stack trace. Until Api validation is enforced, we can assume these errors are truly exceptional so the `error` log level would be the correct choice as opposed to `info` or similar.

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java (hunk 1 above)
Comment (BSaunders95) [reply src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java]: As discussed, we'll address this in a meeting with everyone. I'm of the opinion that we should be following the 'throw early, catch late' principle, but I'm happy to put it to the team

**Pull request merged**